subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -573,13 +573,16 @@ int bt_mesh_sensor_cli_desc_all_get(struct bt_mesh_sensor_cli *cli,
 
 	struct list_rsp list_rsp = {
 		.sensors = sensors,
-		.count = *count,
+		.count = count ? *count : 0,
 	};
 
 	err = model_ackd_send(cli->model, ctx, &msg, sensors ? &cli->ack_ctx : NULL,
 			      BT_MESH_SENSOR_OP_DESCRIPTOR_STATUS, &list_rsp);
 
-	*count = list_rsp.count;
+	if (sensors) {
```

Comment:
```suggestion
	if (count) {
```
?

`count` could have been `NULL`, while `sensor` wasn't

Reply:
I'm not sure. `bt_mesh_sensor_cli_desc_all_get` description says that `count` represents number of elements in `sensors`.